app/component/summary/CustomizeSearch.js

```diff
@@ -0,0 +1,263 @@
+import React from 'react';
+import Slider from '../util/Slider';
+import ToggleButton from '../util/toggle-button';
+import ModeFilter from '../util/mode-filter';
+import * as ItinerarySearchActions from '../../action/ItinerarySearchActions';
+import Select from '../util/select';
+import config from '../../config';
+import { intlShape } from 'react-intl';
+import range from 'lodash/range';
+
+class CustomizeSearch extends React.Component {
+
+  static contextTypes = {
+    getStore: React.PropTypes.func.isRequired,
+    executeAction: React.PropTypes.func.isRequired,
+    intl: intlShape.isRequired,
+  };
+
+  static propTypes = {
+    open: React.PropTypes.bool,
+  };
+
+  constructor() {
+    super();
+    this.onChange = this.onChange.bind(this);
+  }
+
+  componentDidMount() {
+    return this.context.getStore('ItinerarySearchStore').addChangeListener(this.onChange);
+  }
+
+  componentWillUnmount() {
+    return this.context.getStore('ItinerarySearchStore').removeChangeListener(this.onChange);
+  }
+
+  onChange() {
+    return this.forceUpdate();
+  }
+
+  getTicketOptions() {
+    const options = this.context.getStore('ItinerarySearchStore').getTicketOptions();
+
+    return options.map(
+      (option, index) => <option key={index} value={option.value}>{option.displayName}</option>
+    );
+  }
+
+  getSliderStepsArray(min, max, defaultValue, stepCount = 20) {
+    const denom = stepCount / 2;
+    const lowStep = (defaultValue - min) / denom;
+    const lowRange = range(min, defaultValue, lowStep);
+    const highStep = (max - defaultValue) / denom;
+    const highRange = range(defaultValue, max, highStep);
+    return lowRange.concat(highRange.concat(max));
+  }
+
+  getWalkReluctanceSlider = () => {
+    // TODO: connect to this.context.getStore('ItinerarySearchStore').getWalkReluctance()
+
+    const walkReluctanceSliderValues = this.getSliderStepsArray(0.8, 10, 2).reverse();
+
+    return (<section className="offcanvas-section">
+      <Slider
+        headerText={this.context.intl.formatMessage({
+          id: 'walking',
+          defaultMessage: 'Walking',
+        })}
+        defaultValue={10}
+        onSliderChange={e =>
+          this.context.executeAction(
+            ItinerarySearchActions.setWalkReluctance, walkReluctanceSliderValues[e.target.value])}
+        min={0}
+        max={20}
+        step={1}
+        minText={this.context.intl.formatMessage({
+          id: 'avoid-walking',
+          defaultMessage: 'Avoid walking',
+        })}
+        maxText={this.context.intl.formatMessage({
+          id: 'prefer-walking',
+          defaultMessage: 'Prefer walking',
+        })}
+      />
+    </section>);
+  }
+
+  getWalkBoardCostSlider = () => {
+    // TODO: connect to this.context.getStore('ItinerarySearchStore').getWalkBoardCost()
+
+    const walkBoardCostSliderValues = this.getSliderStepsArray(1, 1800, 600).reverse();
```

Comment:
Otp does not allow decimals

at least there are problems when transfermargin is none and transfers slider is moved slightly
```
query Queries {
  
  viewer {
    
    ...F1

  }

}
fragment F0 on Itinerary {
  
  endTime,
  
  legs {
    
    distance,
    
    duration,
    
    endTime,
    
    mode,
    
    realTime,
    
    rentedBike,
    
    route {
      
      id
,
      
      shortName
    }
,
    
    startTime,
    
    transitLeg
  }
,
  
  startTime,
  
  walkDistance
}
fragment F1 on QueryType {
  
  _plan1rkFyF:plan(fromPlace:"Ratamestarinkatu 6, Helsinki::60.199233068750004,24.9398103375",toPlace:"Kamppi, Helsinki::60.168992,24.932366",numItineraries:3,modes:"BUS,TRAM,RAIL,SUBWAY,FERRY,WALK",date:"2016-08-11",time:"13:24:35",walkReluctance:2.000099,walkBoardCost:360.4,minTransferTime:0,walkSpeed:1.200099,maxWalkDistance:2500.1,wheelchair:false,disableRemainingWeightHeuristic:false,arriveBy:false) {
    
    itineraries {
      
      ...F0
,
      
      duration,
      
      endTime,
      
      fares {
        
        cents
,
        
        currency,
        
        type
      },
      
      legs {
        
        agency {
          
          id
,
          
          name
        },
        
        distance,
        
        duration,
        
        endTime,
        
        from {
          
          bikeRentalStation {
            
            id
,
            
            stationId
          },
          
          lat,
          
          lon,
          
          name,
          
          stop {
            
            code,
            
            gtfsId,
            
            id

          }
,
          
          vertexType
        },
        
        intermediateStops {
          
          code,
          
          gtfsId,
          
          id
,
          
          lat,
          
          lon,
          
          name
        },
        
        legGeometry {
          
          length,
          
          points

        },
        
        mode,
        
        mode,
        
        realTime,
        
        rentedBike,
        
        route {
          
          id
,
          
          shortName
        },
        
        startTime,
        
        to {
          
          bikeRentalStation {
            
            id
,
            
            stationId
          },
          
          lat,
          
          lon,
          
          name,
          
          stop {
            
            code,
            
            gtfsId,
            
            id

          }
,
          
          vertexType
        },
        
        transitLeg,
        
        trip {
          
          gtfsId,
          
          id
,
          
          tripHeadsign
        }

      },
      
      startTime,
      
      walkDistance
    }

  }

}
```